lib/portage/package/ebuild/prepare_build_dirs.py

```diff
@@ -484,8 +484,15 @@ def _prepare_fake_filesdir(settings):
 
     # Copy files from real directory to ebuild directory (without metadata).
     if os.path.isdir(real_filesdir) and not os.path.isdir(filesdir):
-        shutil.copytree(
-            real_filesdir, filesdir, copy_function=copyfile
+        shutil.copytree(real_filesdir, filesdir, copy_function=copyfile)
+        apply_recursive_permissions(
+            filesdir,
+            uid=portage_uid,
+            gid=portage_gid,
+            dirmode=0o700,
```

Comment:
Maybe we could open these files to the Portage group at least? I suppose it would ease debugging.

Reply:
Sure, updated to use `dirmode=0o750` and `filemode=0o640`.